Bio/SeqIO/PdbIO.py

```diff
@@ -283,28 +283,16 @@ def PdbAtomIterator(handle):
     # Only import PDB when needed, to avoid/delay NumPy dependency in SeqIO
     from Bio.PDB import PDBParser
 
-    from Bio.File import UndoHandle
-
     with as_handle(handle, "rU") as handle:
-        undo_handle = UndoHandle(handle)
-        firstline = undo_handle.peekline()
-        # check if file is empty
-        if not firstline:
-            raise ValueError("Empty file.")
-
-        # Deduce the PDB ID from the PDB header
-        # ENH: or filename?
-        if firstline.startswith("HEADER"):
-            pdb_id = firstline[62:66]
-        else:
+        struct = PDBParser().get_structure(None, handle)
+        pdb_id = struct.header['idcode']
+        if not pdb_id:
             warnings.warn(
-                "First line is not a 'HEADER'; can't determine PDB ID. "
-                "Line: %r" % firstline,
+                "'HEADER' line not found; can't determine PDB ID.",
```

Comment:
This warning message seems far less useful for debugging a failure.

Reply:
The PDB parser does not require that the first line contains the HEADER. It should, but it doesn't. From the viewpoint of the PDB parser, there is no specific line that can be referred to here.

Reply:
And since the old warning just included the first line, that is very easy for the user to inspect themselves at the command line or otherwise?